panic: in loadDebugInfoMapsCompileUnit: interface {} is nil, not dwarf.Offset

```
Delve Debugger
Version: 1.2.0
Build: $Id: 068e2451004e95d0b042e5257e34f0f08ce01466 $
go version go1.12.7 linux/amd64
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.3 LTS
Release:	16.04
Codename:	xenial
```

(this is an instance on AWS)

Trying to attach to a `cockroach` process:

```
$ dlv attach 24490
panic: interface conversion: interface {} is nil, not dwarf.Offset

goroutine 13 [running]:
github.com/go-delve/delve/pkg/proc.(*BinaryInfo).loadDebugInfoMapsCompileUnit(0xc000001800, 0xc0000c60e0, 0xc0000fe6e0, 0xc0000ac480, 0xc0147b5950)
	/home/ubuntu/go/src/github.com/go-delve/delve/pkg/proc/types.go:414 +0x1861
github.com/go-delve/delve/pkg/proc.(*BinaryInfo).loadDebugInfoMaps(0xc000001800, 0xc0000fe6e0, 0xc0069fc000, 0xcb4b4b, 0xcb4b4b, 0xc0000b6bb0, 0x0)
	/home/ubuntu/go/src/github.com/go-delve/delve/pkg/proc/types.go:267 +0x591
created by github.com/go-delve/delve/pkg/proc.loadBinaryInfoElf
	/home/ubuntu/go/src/github.com/go-delve/delve/pkg/proc/bininfo.go:904 +0x3d5
```

Hopefully repros for others via this binary: https://drive.google.com/file/d/1-CkILyxXwzjTBLIvzCJ21-ud6vuBHs6w/view?usp=sharing (definitely repros here)

```
dlv exec ./cockroach.boom version
```

I worked around this via the following diff:

```diff
diff --git a/pkg/proc/types.go b/pkg/proc/types.go
index b560d29..017ebc8 100644
--- a/pkg/proc/types.go
+++ b/pkg/proc/types.go
@@ -411,7 +411,8 @@ func (bi *BinaryInfo) loadDebugInfoMapsCompileUnit(ctxt *loadDebugInfoMapsContex
                                                break
                                        }
                                        if entry.Tag == dwarf.TagInlinedSubroutine {
-                                               originOffset := entry.Val(dwarf.AttrAbstractOrigin).(dwarf.Offset)
+                                               originOffset, ok := entry.Val(dwarf.AttrAbstractOrigin).(dwarf.Offset)
+                                               if ok {
                                                name := ctxt.abstractOriginNameTable[originOffset]
                                                if ranges, _ := image.dwarf.Ranges(entry); len(ranges) == 1 {
                                                        ok1 = true
@@ -431,6 +432,9 @@ func (bi *BinaryInfo) loadDebugInfoMapsCompileUnit(ctxt *loadDebugInfoMapsContex
                                                                Parent:   &fn,
                                                        })
                                                }
+                                               } else {
+                                                       fmt.Println("would've crashed")
+                                               }
                                        }
                                        reader.SkipChildren()
                                }
```

and was able to use dlv normally. (It printed the message dozens of times during startup).

As an additional clue, the issue does not occur with a binary that I thought would be similar:

https://binaries.cockroachdb.com/cockroach-v19.2.0-alpha.20190701.linux-amd64.tgz

--- a/pkg/dwarf/dwarfbuilder/info.go
+++ b/pkg/dwarf/dwarfbuilder/info.go
@@ -92,7 +92,9 @@ func (b *Builder) TagOpen(tag dwarf.Tag, name string) dwarf.Offset {
 	ts.tag = tag
 	b.info.WriteByte(0)
 	b.tagStack = append(b.tagStack, ts)
-	b.Attr(dwarf.AttrName, name)
+	if name != "" {
+		b.Attr(dwarf.AttrName, name)
+	}
 
 	return ts.off
 }

--- a/pkg/proc/dwarf_expr_test.go
+++ b/pkg/proc/dwarf_expr_test.go
@@ -292,3 +292,18 @@ func TestLocationCovers(t *testing.T) {
 	}
 
 }
+
+func TestIssue1636_InlineWithoutOrigin(t *testing.T) {
+	// Gcc (specifically GNU C++11 6.3.0) will emit DW_TAG_inlined_subroutine
+	// without a DW_AT_abstract_origin or a name. What is an inlined subroutine
+	// without a reference to an abstract origin or even a name? Regardless,
+	// Delve shouldn't crash.
+	dwb := dwarfbuilder.New()
+	dwb.AddCompileUnit("main", 0x0)
+	dwb.AddSubprogram("main.main", 0x40100, 0x41000)
+	dwb.TagOpen(dwarf.TagInlinedSubroutine, "")
+	dwb.TagClose()
+	dwb.TagClose()
+	dwb.TagClose()
+	fakeBinaryInfo(t, dwb)
+}

--- a/pkg/proc/types.go
+++ b/pkg/proc/types.go
@@ -410,7 +410,7 @@ func (bi *BinaryInfo) loadDebugInfoMapsCompileUnit(ctxt *loadDebugInfoMapsContex
 					if entry.Tag == 0 {
 						break
 					}
-					if entry.Tag == dwarf.TagInlinedSubroutine {
+					if entry.Tag == dwarf.TagInlinedSubroutine && entry.Val(dwarf.AttrAbstractOrigin) != nil {
 						originOffset := entry.Val(dwarf.AttrAbstractOrigin).(dwarf.Offset)
 						name := ctxt.abstractOriginNameTable[originOffset]
 						if ranges, _ := image.dwarf.Ranges(entry); len(ranges) == 1 {

